samza-yarn/src/main/java/org/apache/samza/util/hadoop/CertFSFileSystem.java

```diff
@@ -0,0 +1,461 @@
+    } else {
+      bcstyleX500Name = bcstyleX500NameRaw;
+    }
+    return bcstyleX500Name;
+  }
+
+  /**
+   * create the certificate signing request with the giving {@link BCStyle} {@link X500Name}
+   * @param bcstyleX500Name in String format, e.g."CN=hostname.com, OU=Security, O=LinkedIn, L=Mountain View, ST=California, C=US"
+   * @return Certificate signing request in string format
+   * @throws CertFSException certificate local signing issue
+   * @throws IOException reader/writer related exception
+   */
+  private String createCSR(final String bcstyleX500Name) throws CertFSException, IOException {
+    X500Name name = new X500Name(bcstyleX500Name);
+    PKCS10CertificationRequestBuilder pkcs10CertificationRequestBuilder = new JcaPKCS10CertificationRequestBuilder(
+        name, _keyPair.getPublic()
+    );
+    JcaContentSignerBuilder contentSignerBuilder = new JcaContentSignerBuilder(CertFSConstants.SIGNATURE_ALGORITHM);
+    ContentSigner signer;
+    try {
+      signer = contentSignerBuilder.build(_keyPair.getPrivate());
+    } catch (OperatorCreationException e) {
+      throw new CertFSException("cannot create content signer for csr with private key");
+    }
+    PKCS10CertificationRequest csr = pkcs10CertificationRequestBuilder.build(signer);
+
+    StringWriter sw = new StringWriter();
+    JcaPEMWriter pemWriter = new JcaPEMWriter(sw);
+    pemWriter.writeObject(csr);
+    pemWriter.close();
+    return sw.toString();
+  }
+
+  /**
+   * construct and return the private key in pem format from the keyPair
+   * @return String
+   */
+  private String getPrivateKeyPem() {
+    // The stream is already base64 encoded.
+    byte[] privateKey = _keyPair.getPrivate().getEncoded();
+
+    Base64.Encoder encoder = Base64.getMimeEncoder(64, new byte[] {'\r', '\n'});
+    String privateKeyPem = String.format(
+        "%s\r\n%s\r\n%s",
+        "-----BEGIN RSA PRIVATE KEY-----",
+        encoder.encodeToString(privateKey),
+        "-----END RSA PRIVATE KEY-----"
+    );
+    return privateKeyPem;
+  }
+
+  /**
+   * get https client based on the configuration settings
+   * @return HttpClient
+   * @throws CertFSException when there is any exception when creating httpClient for SSL
+   */
+  private HttpClient getHttpsClient() throws CertFSException {
+    String trustStoreType = getConf().get(CertFSConstants.TRUSTSTORE_TYPE, "JKS");
+    String keyStoreType = getConf().get(CertFSConstants.KEYSTORE_TYPE, "PKCS12");
+    try {
+      KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+      KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+
+      String trustStoreLocation = getConf().get(CertFSConstants.TRUSTSTORE_LOCATION, "");
+      log.info("trust store location is: {}", trustStoreLocation);
+      String trustStorePassword = getConf().get(CertFSConstants.TRUSTSTORE_PASSWORD, "");
+      trustStore.load(new FileInputStream(trustStoreLocation), trustStorePassword.toCharArray());
```

Comment:
Thanks. Will throw exception when the location is empty. But I am considering to default to use empty password if password is empty.